tests/integration/it_table_shapes_test.py

```diff
@@ -51,13 +51,16 @@ def test_filtered_list(client, db_session):
     #  table_shapes = TableShapes(client, filter_by=["description=B"]
     #  assert len(table_shapes) == 1
     #  assert iter(table_shapes).next().id == 1
-    # @todo #260:30min Uncomment below assertion and fix the test once #273 is fixed.
-    #  Currently there is an issue with template caching which is reported by #273 1st point. Lets uncomment
-    #  below assertion checking if filter logic works as expected by checking table shapes rendered within list route.
-
-    # html = response.data.decode('utf-8')
-    # assert html.count('<article class="table_shape">') == 1
-    # assert html.count('<a class="action" href="/table_shapes/edit/1">Edit</a>') == 1
+    # @todo #315:30min Fix cache problem which prints the table twice in list.
+    #  This problem was supposed to be corrected in #273 1st point, but for
+    #  some reason filtered shape list is still printing garbage to output.
+    #  After correcting this problem, uncomment the tests below.
```

Comment:
@paulodamaso Why did you decide that problem with cache do not resolved ?

Reply:
@vladarefiev because it's resolved :) I've spent 20 sec to find the template and realize that problem there https://github.com/timelesslounge/timelessis/blob/master/timeless/templates/restaurants/table_shapes/list.html#L8

Reply:
@vladarefiev Because the test still fails

Reply:
@vryazanov I dopn't see any problem in that line, sorry

Reply:
@paulodamaso Problem with was fixed by @ammaratef45 here https://github.com/timelesslounge/timelessis/blob/master/tests/conftest.py#L77

Reply:
@paulodamaso Your problem does not related to cache.
@vryazanov noticed that tag `acticle` is situated inside loop, so here is you have broken `assert html.count("<article class=\"table_shape\">") == 1`

Reply:
@vladarefiev Sorry I can't understand why it's wrong. We should show all the shapes that comply with some parameter (the filter) and thios result can be a list, `article` is correct where it is. I thought that the problem was about cache because the issue is happening even fith the cache problem fix.

Reply:
@paulodamaso May be, I don't know, I just want to say that issue with cache is fixed.